src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/CassandraEventSchema.scala

```diff
@@ -0,0 +1,173 @@
+    for {
+      kw <- item.computedfeatures.keywords
+      tileid <- TileUtils.tile_seq_from_places(item.computedfeatures.places)
+      periodType <- Utils.getCassandraPeriodTypes
+    } yield PopularTopicAggregate(
+      pipelinekey = item.pipelinekey,
+      periodstartdate = Period(item.eventtime, periodType).startTime(),
+      periodenddate = Period(item.eventtime, periodType).endTime(),
+      periodtype = periodType.periodTypeName,
+      period = periodType.format(item.eventtime),
+      tilex = tileid.column,
+      tiley = tileid.row,
+      tilez = tileid.zoom,
+      topic = kw,
+      externalsourceid = item.externalsourceid,
+      mentioncount = item.computedfeatures.mentions,
+      avgsentimentnumerator = 0,
+      avgsentiment = item.computedfeatures.sentiment.neg_avg
+    )
+  }
+}
+
+object CassandraEventTopicSchema {
+  def apply(item: Event): Seq[EventTopics] = {
+    for {
+      kw <- item.computedfeatures.keywords
+    } yield EventTopics(
+      pipelinekey = item.pipelinekey,
+      eventids = Seq(item.eventid),
+      topic = kw.toLowerCase,
+      insertiontime = new Date().getTime,
+      externalsourceid = item.externalsourceid
+    )
+  }
+}
+
+object CassandraEventPlacesSchema {
+  def apply(item: Event): Seq[EventPlaces] = {
+    for {
+      ct <- Utils.getConjunctiveTopics(Option(item.computedfeatures.keywords))
+      location <- item.computedfeatures.places
+    } yield EventPlaces(
+      pipelinekey = item.pipelinekey,
+      centroidlat = location.centroidlat,
+      centroidlon = location.centroidlon,
+      eventids = Seq(item.eventid),
+      conjunctiontopic1 = ct._1,
+      conjunctiontopic2 = ct._2,
+      conjunctiontopic3 = ct._3,
+      insertiontime = new Date().getTime,
+      externalsourceid = item.externalsourceid,
+      placeid = location.placeid
+    )
+  }
+}
+
+object Utils {
+  private val ConjunctiveTopicComboSize = 3
+  private val DefaultPrimaryLanguage = "en"//TODO thread the site settings primary language to getConjunctiveTopics
+
+  def getCassandraPeriodTypes: Seq[PeriodType] = {
+    Seq(PeriodType.Day, PeriodType.Hour, PeriodType.Month, PeriodType.Week, PeriodType.Year)
+  }
+
+  def getConjunctiveTopics(topicSeq: Option[Seq[String]], langcode: Option[String] = None): Seq[(String, String, String)] = {
+    topicSeq match {
+      case Some(topics) =>
+        (topics ++ Seq("", "")).toList.combinations(ConjunctiveTopicComboSize).toList.map(combo => {
+          val sortedCombo = combo.sortWith{(a, b) =>
+            Ordering.comparatorToOrdering(Collator.getInstance(new Locale(langcode.getOrElse(langcode.getOrElse(DefaultPrimaryLanguage))))).compare(a,b) < 0 && a != ""
+          }
+
+          (sortedCombo(0), sortedCombo(1), sortedCombo(2))
+        })
+      case None => Seq()
+    }
+  }
+
+  def getSentimentScore(sentiments: List[Double]): Float = {
+    Option(sentiments) match {
+      case None => 0F
+      case Some(sentimentList) => {
+        var neg_sent = 0F
```

Comment:
This may be clearer with a ternary as it'd avoid the unidiomatic var:

```scala
if (sentimentList.isEmpty) 0F else sentimentList.head.toFloat
```

That being said: why the need to cast this to float instead of changing the sentiment field in the database to a double? That would avoid us loosing precision.